app/src/main/java/com/android/unio/model/association/AssociationViewModel.kt

```diff
@@ -23,22 +29,40 @@ class AssociationViewModel(val repository: AssociationRepository) : ViewModel()
       _associationsByCategory
 
   init {
-    repository.init { getAssociations() }
+    associationRepository.init { getAssociations() }
   }
 
   companion object {
     val Factory: ViewModelProvider.Factory =
         object : ViewModelProvider.Factory {
           @Suppress("UNCHECKED_CAST")
           override fun <T : ViewModel> create(modelClass: Class<T>): T {
-            return AssociationViewModel(AssociationRepositoryFirestore(Firebase.firestore)) as T
+            return AssociationViewModel(
+                AssociationRepositoryFirestore(Firebase.firestore),
+                EventRepositoryFirestore(Firebase.firestore))
+                as T
           }
         }
   }
 
+  fun getEventsForAssociation(association: Association, onSuccess: (List<Event>) -> Unit) {
```

Comment:
Event though it is pretty obvious what is does by the name, this function (and others) should still be commented with a bit of javadoc